src/utils/monitoring-sites-data.js

```diff
@@ -111,6 +130,46 @@ const setRiverFlowSitesInBoundingBox = async (map, corners) => {
   map.getSource("riverFlowSites").setData(sites);
 };
 
+const setBristolWaterQualitySitesInBoundingBox = async (map) => {
+  const query = bristolWaterQualityQuery(map);
+  const response = await fetch(query).then((response) => response.json());
+  const records = response.records;
+  const siteIds = response.facet_groups[0].facets.map((facet) => facet.name);
+
+  const latestRecords = [];
+  for (const siteId of siteIds) {
+    let latestRecord = records.find(
+      (record) => record.fields.siteid === siteId
+    );
+
+    if (latestRecord) {
+      latestRecords.push(latestRecord);
+    } else {
+      const recordsForSiteQuery = getRecordsForBWQSiteQuery(siteId);
+      const response = await fetch(recordsForSiteQuery).then((response) =>
+        response.json()
+      );
+      const record = response.records[0];
+      latestRecords.push(record);
+    }
+  }
+
+  const bristolSites = {
+    type: "FeatureCollection",
+    features: latestRecords.map((record) => ({
+      geometry: record.geometry,
+      properties: {
+        label: record.fields.sitename,
+        siteId: record.fields.siteid,
+        apiUrl: getRecordsForBWQSiteQuery(record.fields.siteid),
+        latestRecordDate: record.fields.datetime.split("T")[0],
```

Comment:
Used to fetch the latest record info for the popup when a user clicks on the site